### ISSUE TITLE
systemd services files are incorrectly installed since the meson switch

### ISSUE THREAD

**Issue by bentiss** [libratbag/ratbagd#18]
While trying to upgrade the fedora rpm:
```
RPM build errors:
    File not found: ~/rpmbuild/BUILDROOT/ratbagd-0.2.20.g44471d3-1.fc27.x86_64/usr/lib/systemd/system/ratbagd.service
    File not found: ~/rpmbuild/BUILDROOT/ratbagd-0.2.20.g44471d3-1.fc27.x86_64/usr/share/dbus-1/system-services/org.freedesktop.ratbag1.service
```

Looks like `meson.build` doesn't replace the `.in` file properly.

I quickly got something like:
```
diff --git a/meson.build b/meson.build
index e2abdac..ea7b0ce 100644
--- a/meson.build
+++ b/meson.build
@@ -119,20 +119,22 @@ if unitdir == ''
 endif
 
 config_bindir = configuration_data()
-config_bindir.set('bindir', get_option('bindir'))
+config_bindir.set('bindir', join_paths (get_option ('prefix'), get_option ('bindir')))
 
-install_data('ratbagd.service.in',
-            configuration : config_bindir,
-            install_dir : unitdir)
+configure_file(input : 'ratbagd.service.in',
+              output : 'ratbagd.service',
+              configuration : config_bindir,
+              install_dir : unitdir)
 
 dbusdir = get_option('dbus-root-dir')
 if dbusdir == ''
        dbusdir = get_option('datadir') + '/dbus-1'
 endif
 
-install_data('dbus/org.freedesktop.ratbag1.service.in',
-            configuration : config_bindir,
-            install_dir : dbusdir + '/system-services')
+configure_file(input : 'dbus/org.freedesktop.ratbag1.service.in',
+              output : 'org.freedesktop.ratbag1.service',
+              configuration : config_bindir,
+              install_dir : dbusdir + '/system-services')
 install_data('dbus/org.freedesktop.ratbag1.conf',
             configuration : config_bindir,
             install_dir : dbusdir + '/system.d')
```

But I am not sure about the `join_paths` while setting config_bindir, in case `bindir` points to an absolute path.

Offloading this here in case some meson guru sees it :-)

**Comment by Hjdskes** [comment]
Hi @bentiss,

I'm not a Meson guru, but this seems like a nice issue to participate in during the community bonding period! I'm one of this year's GSoC students and I'll be rewriting Piper this summer under @whot's guidance.

I read up on Meson a bit and I looked how other projects seem to tackle this issue. GNOME's Polari [seems to do what you are suggesting](https://github.com/GNOME/polari/blob/b725582b86a28a782c5984a6de97cbf574569bd9/data/meson.build#L21) (with `bindir` defined [here](https://github.com/GNOME/polari/blob/b9d575c1c7e56fcc973239caf6134ab32a32cb15/meson.build#L13)). GNOME Recipes instead [hardcodes](https://git.gnome.org/browse/recipes/tree/meson.build#n85) `bindir` to `$prefix/bin` but it does use the same solution you suggest for its other paths, [example](https://git.gnome.org/browse/recipes/tree/meson.build#n80) (I'm not sure why they use an array, though).

**Comment by whot** [comment]
we should use join_paths() where a component may be absolute because meson takes care of the rest. e.g. join_path("/foo", "/bar/bin") will ignore the /foo component. so for bindir, join_path() is recommended.
and for consistency, we should use it for the dbusdir bits too, thought it matters less there. 

http://mesonbuild.com/Reference-manual.html#join_paths

@Hjdskes, want to take that one and submit a PR for it?

**Comment by Hjdskes** [comment]
I will have something ready within a few hours!

**Comment by Hjdskes** [comment]
See #20.

**Comment by whot** [comment]
Fixed with PR #20, thanks
